lib/java/polyglot-ydoc-server/src/main/java/org/enso/polyfill/encoding/EncodingPolyfill.java

```diff
@@ -0,0 +1,56 @@
+package org.enso.polyfill.encoding;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+import org.enso.polyfill.Polyfill;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+import org.graalvm.polyglot.proxy.ProxyExecutable;
+
+public final class EncodingPolyfill implements ProxyExecutable, Polyfill {
+
+    private static final String TEXT_DECODER_DECODE = "text-decoder-decode";
+
+    private static final String ENCODING_POLYFILL_JS = "encoding-polyfill.js";
+
+    public EncodingPolyfill() {
+    }
+
+    @Override
+    public void initialize(Context ctx) {
+        Source encodingPolyfillJs = Source
+                .newBuilder("js", EncodingPolyfill.class.getResource(ENCODING_POLYFILL_JS))
+                .buildLiteral();
+
+        ctx.eval(encodingPolyfillJs).execute(this);
+    }
+
+    @Override
+    public Object execute(Value... arguments) {
+        var command = arguments[0].asString();
+        System.err.println(command + " " + Arrays.toString(arguments));
+
+        return switch (command) {
+            case TEXT_DECODER_DECODE -> {
+                var encoding = arguments[1].asString();
+                var data = arguments[2].as(int[].class);
+
+                var charset = encoding == null ? StandardCharsets.UTF_8 : Charset.forName(encoding);
+                // Convert unsigned Uint8Array to byte[]
```

Comment:
Try to obtain `.as(ByteBuffer.class)` - I've [heard it should be possible](https://graalvm.slack.com/archives/CN9L59P61/p1695307910431519?thread_ts=1695300535.788289&cid=CN9L59P61) to get it from `Uint8Array`.

Reply:
The `ByteBuffer` interop currently works only with arrays that are backed by the Java buffers https://github.com/oracle/graaljs/blob/master/graal-js/src/com.oracle.truffle.js.test/src/com/oracle/truffle/js/test/interop/InteropByteBufferTest.java